resources/config/varnish/fos_purge_xkey.vcl

```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import xkey;
+
+sub fos_purge_recv {
+    if (req.method == "PURGE") {
+        if (!client.ip ~ invalidators) {
+            return (synth(405, "Not allowed"));
+        }
+
+        if (req.http.key) {
+            # If possible use soft purge (expiry) to respect grace time instead of normal purge.
+            # set req.http.n-gone = xkey.softpurge(req.http.key);
```

Comment:
our aim is that these snippets can be deployed to the server and used as is. i would prefer to use different http methods to distinguish. that would also allow the application to use both modes depending on what is happening.